src/simulations/hedge.py

```diff
@@ -40,29 +41,31 @@ def __init__(
         factors0: Optional[dict] = None,
         clusters0: Optional[dict] = None,
         prm: Optional[dict] = None,
-        other_prm: Optional[dict] = None
+        other_prm: Optional[dict] = None,
     ):
         """Initialise HEDGE object and initial properties."""
         # update object properties
         self.labels_day = ["wd", "we"]
         self.n_homes = n_homes
         self.homes = range(self.n_homes)
-
         # load input data
         self._load_input_data(prm, other_prm, factors0, clusters0)
 
-    def _replace_other_prm(self, prm, other_prm):
-        prm = prm.copy()
+    def _replace_car_prm(self, prm, other_prm):
+        prm = copy.deepcopy(prm)
```

Comment:
@julie-vienne fyi this was the main problem, that there there was a shallow copy instead of a deep copy, so when we edited the car info in the `passive_hedge` this also edited the main `prm`